Source/Evolve/WarpXEvolve.cpp

```diff
@@ -321,6 +321,10 @@ WarpX::Evolve (int numsteps)
             reduced_diags->ComputeDiags(step);
             reduced_diags->WriteToFile(step);
         }
+
+        // Calling backtransform diagnstics, passing step, force-flush=false, back-transform = true
+        multi_diags->FilterComputePackFlush( step, false, true );
+        // Calling regular diagnostics
         multi_diags->FilterComputePackFlush( step );
```

Comment:
This becomes very complicated here as the multi_diags contain:
- reduced diags
- full diags
  - lab frame
  - BTD

I would say we implement all logic switches inside the function and avoid an additional sate split here? :)

Reply:
I agree -- and would be good to discuss this. The same logic has gone in #1898 actually to separate the called for BTD and regular diags